lib/src/sdk/flutter.dart

```diff
@@ -20,13 +20,45 @@ class FlutterSdk extends Sdk {
   @override
   Version get firstPubVersion => Version.parse('1.19.0');
 
-  static final bool _isAvailable =
-      Platform.environment.containsKey('FLUTTER_ROOT');
-  static final String _rootDirectory = Platform.environment['FLUTTER_ROOT'];
+  static final bool _isAvailable = _rootDirectory != null;
+  static final String _rootDirectory = () {
+    // If FLUTTER_ROOT is specified, then this always points to the Flutter SDK
+    if (Platform.environment.containsKey('FLUTTER_ROOT')) {
+      return Platform.environment['FLUTTER_ROOT'];
+    }
+
+    // We can try to find the Flutter SDK relative to the Dart SDK.
+    // We know that the Dart SDK is always present, this is found relative to
+    // the `dart` executable, for details see: lib/src/sdk/dart.dart
+    //
+    // Once we have the location of the Dart SDK, we can look at its parent
+    // directories, if going 3 levels-up and down `bin/cache/dart-sdk/` is equal
+    // to the Dart SDK root, then it's probably because we are located inside
+    // the Flutter SDK, at: `$FLUTTER_ROOT/bin/cache/dart-sdk`
+    final parts = p.split(sdk.rootDirectory);
+    if (parts.length > 3) {
+      // Go 3-levels up from the Dart SDK root
+      final flutterSdk = p.joinAll(parts.take(parts.length - 3));
+      // If going down 'bin/cache/dart-sdk/' yields the same path as the Dart
+      // SDK has, then it's probably because the Dart SDK is located inside
+      // the Flutter SDK.
+      final dartRootFromFlutterSdk = p.join(
+        flutterSdk,
+        'bin',
+        'cache',
+        'dart-sdk',
+      );
+      if (p.equals(sdk.rootDirectory, dartRootFromFlutterSdk)) {
+        return flutterSdk;
+      }
+    }
+
+    return null;
+  }();
 
   @override
   String get installMessage =>
-      'Flutter users should run `flutter pub get` instead of `pub get`.';
+      'Flutter users should run `flutter pub get` instead of `dart pub get`.';
```

Comment:
Is this still true?

Reply:
Yes, because `flutter pub get` has some hooks generating code for plugins.

But I think you'll only really see this message if the Flutter SDK isn't found.